### ISSUE TITLE
Failed to generate thumbnail for WebP image format

### ISSUE THREAD

**Issue by gorjan-mishevski** [saleor/saleor#7432]
### What I'm trying to achieve
Upload a product image

### Steps to reproduce the problem

1. Product
2. Add media
3. Open media window, the thumbnails are broken.
4. Storefront thumbnails are broken.

### What I expected to happen
Working thumbnail version of webp image format.

### Screenshots and logs

```
{"asctime": "2021-06-05T14:55:25Z", "levelname": "ERROR", "lineno": 118, "message": "Thumbnail generation failed", "name": "versatileimagefield.image_warmer", "pathname": "/usr/local/lib/python3.9/site-packages/versatileimagefield/image_warmer.py", "process": 8, "threadName": "ThreadPoolExecutor-1_0", "exc_info": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/image_warmer.py\", line 114, in _prewarm_versatileimagefield\n    url = get_url_from_image_key(versatileimagefieldfile, size_key)\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/utils.py\", line 196, in get_url_from_image_key\n    img_url = img_url[size_key].url\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/datastructures/sizedimage.py\", line 141, in __getitem__\n    self.create_resized_image(\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/datastructures/sizedimage.py\", line 202, in create_resized_image\n    imagefile = self.process_image(\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/versatileimagefield.py\", line 173, in process_image\n    image.save(\n  File \"/usr/local/lib/python3.9/site-packages/PIL/Image.py\", line 2172, in save\n    save_handler(self, fp, filename)\n  File \"/usr/local/lib/python3.9/site-packages/PIL/WebPImagePlugin.py\", line 327, in _save\n    data = _webp.WebPEncode(\nTypeError: a bytes-like object is required, not 'NoneType'", "path": "products/web_4017645043138_30067_Aktiv_Glatt_Feuchtigkeitsserum_FL_side.webp", "hostname": "ubuntu-s-1vcpu-1gb-ams3-01"}
{"asctime": "2021-06-05T14:55:25Z", "levelname": "ERROR", "lineno": 118, "message": "Thumbnail generation failed", "name": "versatileimagefield.image_warmer", "pathname": "/usr/local/lib/python3.9/site-packages/versatileimagefield/image_warmer.py", "process": 8, "threadName": "ThreadPoolExecutor-1_0", "exc_info": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/image_warmer.py\", line 114, in _prewarm_versatileimagefield\n    url = get_url_from_image_key(versatileimagefieldfile, size_key)\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/utils.py\", line 196, in get_url_from_image_key\n    img_url = img_url[size_key].url\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/datastructures/sizedimage.py\", line 141, in __getitem__\n    self.create_resized_image(\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/datastructures/sizedimage.py\", line 202, in create_resized_image\n    imagefile = self.process_image(\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/versatileimagefield.py\", line 173, in process_image\n    image.save(\n  File \"/usr/local/lib/python3.9/site-packages/PIL/Image.py\", line 2172, in save\n    save_handler(self, fp, filename)\n  File \"/usr/local/lib/python3.9/site-packages/PIL/WebPImagePlugin.py\", line 327, in _save\n    data = _webp.WebPEncode(\nTypeError: a bytes-like object is required, not 'NoneType'", "path": "products/web_4017645043138_30067_Aktiv_Glatt_Feuchtigkeitsserum_FL_side.webp", "hostname": "ubuntu-s-1vcpu-1gb-ams3-01"}
{"asctime": "2021-06-05T14:55:25Z", "levelname": "ERROR", "lineno": 118, "message": "Thumbnail generation failed", "name": "versatileimagefield.image_warmer", "pathname": "/usr/local/lib/python3.9/site-packages/versatileimagefield/image_warmer.py", "process": 8, "threadName": "ThreadPoolExecutor-1_0", "exc_info": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/image_warmer.py\", line 114, in _prewarm_versatileimagefield\n    url = get_url_from_image_key(versatileimagefieldfile, size_key)\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/utils.py\", line 196, in get_url_from_image_key\n    img_url = img_url[size_key].url\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/datastructures/sizedimage.py\", line 141, in __getitem__\n    self.create_resized_image(\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/datastructures/sizedimage.py\", line 202, in create_resized_image\n    imagefile = self.process_image(\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/versatileimagefield.py\", line 173, in process_image\n    image.save(\n  File \"/usr/local/lib/python3.9/site-packages/PIL/Image.py\", line 2172, in save\n    save_handler(self, fp, filename)\n  File \"/usr/local/lib/python3.9/site-packages/PIL/WebPImagePlugin.py\", line 327, in _save\n    data = _webp.WebPEncode(\nTypeError: a bytes-like object is required, not 'NoneType'", "path": "products/web_4017645043138_30067_Aktiv_Glatt_Feuchtigkeitsserum_FL_side.webp", "hostname": "ubuntu-s-1vcpu-1gb-ams3-01"}
{"asctime": "2021-06-05T14:55:25Z", "levelname": "ERROR", "lineno": 118, "message": "Thumbnail generation failed", "name": "versatileimagefield.image_warmer", "pathname": "/usr/local/lib/python3.9/site-packages/versatileimagefield/image_warmer.py", "process": 8, "threadName": "ThreadPoolExecutor-1_0", "exc_info": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/image_warmer.py\", line 114, in _prewarm_versatileimagefield\n    url = get_url_from_image_key(versatileimagefieldfile, size_key)\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/utils.py\", line 196, in get_url_from_image_key\n    img_url = img_url[size_key].url\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/datastructures/sizedimage.py\", line 141, in __getitem__\n    self.create_resized_image(\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/datastructures/sizedimage.py\", line 202, in create_resized_image\n    imagefile = self.process_image(\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/versatileimagefield.py\", line 173, in process_image\n    image.save(\n  File \"/usr/local/lib/python3.9/site-packages/PIL/Image.py\", line 2172, in save\n    save_handler(self, fp, filename)\n  File \"/usr/local/lib/python3.9/site-packages/PIL/WebPImagePlugin.py\", line 327, in _save\n    data = _webp.WebPEncode(\nTypeError: a bytes-like object is required, not 'NoneType'", "path": "products/web_4017645043138_30067_Aktiv_Glatt_Feuchtigkeitsserum_FL_side.webp", "hostname": "ubuntu-s-1vcpu-1gb-ams3-01"}
{"asctime": "2021-06-05T14:55:25Z", "levelname": "ERROR", "lineno": 118, "message": "Thumbnail generation failed", "name": "versatileimagefield.image_warmer", "pathname": "/usr/local/lib/python3.9/site-packages/versatileimagefield/image_warmer.py", "process": 8, "threadName": "ThreadPoolExecutor-1_0", "exc_info": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/image_warmer.py\", line 114, in _prewarm_versatileimagefield\n    url = get_url_from_image_key(versatileimagefieldfile, size_key)\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/utils.py\", line 196, in get_url_from_image_key\n    img_url = img_url[size_key].url\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/datastructures/sizedimage.py\", line 141, in __getitem__\n    self.create_resized_image(\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/datastructures/sizedimage.py\", line 202, in create_resized_image\n    imagefile = self.process_image(\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/versatileimagefield.py\", line 173, in process_image\n    image.save(\n  File \"/usr/local/lib/python3.9/site-packages/PIL/Image.py\", line 2172, in save\n    save_handler(self, fp, filename)\n  File \"/usr/local/lib/python3.9/site-packages/PIL/WebPImagePlugin.py\", line 327, in _save\n    data = _webp.WebPEncode(\nTypeError: a bytes-like object is required, not 'NoneType'", "path": "products/web_4017645043138_30067_Aktiv_Glatt_Feuchtigkeitsserum_FL_side.webp", "hostname": "ubuntu-s-1vcpu-1gb-ams3-01"}
{"asctime": "2021-06-05T14:55:25Z", "levelname": "ERROR", "lineno": 118, "message": "Thumbnail generation failed", "name": "versatileimagefield.image_warmer", "pathname": "/usr/local/lib/python3.9/site-packages/versatileimagefield/image_warmer.py", "process": 8, "threadName": "ThreadPoolExecutor-1_0", "exc_info": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/image_warmer.py\", line 114, in _prewarm_versatileimagefield\n    url = get_url_from_image_key(versatileimagefieldfile, size_key)\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/utils.py\", line 196, in get_url_from_image_key\n    img_url = img_url[size_key].url\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/datastructures/sizedimage.py\", line 141, in __getitem__\n    self.create_resized_image(\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/datastructures/sizedimage.py\", line 202, in create_resized_image\n    imagefile = self.process_image(\n  File \"/usr/local/lib/python3.9/site-packages/versatileimagefield/versatileimagefield.py\", line 173, in process_image\n    image.save(\n  File \"/usr/local/lib/python3.9/site-packages/PIL/Image.py\", line 2172, in save\n    save_handler(self, fp, filename)\n  File \"/usr/local/lib/python3.9/site-packages/PIL/WebPImagePlugin.py\", line 327, in _save\n    data = _webp.WebPEncode(\nTypeError: a bytes-like object is required, not 'NoneType'", "path": "products/web_4017645043138_30067_Aktiv_Glatt_Feuchtigkeitsserum_FL_side.webp", "hostname": "ubuntu-s-1vcpu-1gb-ams3-01"}
{"asctime": "2021-06-05T14:55:25Z", "levelname": "ERROR", "lineno": 113, "message": "Failed to generate thumbnails", "name": "saleor.core.utils", "pathname": "/app/saleor/core/utils/__init__.py", "process": 8, "threadName": "ThreadPoolExecutor-1_0", "paths": ["products/web_4017645043138_30067_Aktiv_Glatt_Feuchtigkeitsserum_FL_side.webp", "products/web_4017645043138_30067_Aktiv_Glatt_Feuchtigkeitsserum_FL_side.webp", "products/web_4017645043138_30067_Aktiv_Glatt_Feuchtigkeitsserum_FL_side.webp", "products/web_4017645043138_30067_Aktiv_Glatt_Feuchtigkeitsserum_FL_side.webp", "products/web_4017645043138_30067_Aktiv_Glatt_Feuchtigkeitsserum_FL_side.webp", "products/web_4017645043138_30067_Aktiv_Glatt_Feuchtigkeitsserum_FL_side.webp"], "hostname": "ubuntu-s-1vcpu-1gb-ams3-01"}
{"asctime": "2021-06-05T14:55:25Z", "levelname": "INFO", "lineno": 122, "message": "Task saleor.product.thumbnails.create_product_thumbnails[ec9d22e8-ed36-40d2-ba6a-5d09b3f1d965] succeeded in 0.17312861210666597s: None", "name": "celery.app.trace", "pathname": "/usr/local/lib/python3.9/site-packages/celery/app/trace.py", "process": 8, "threadName": "ThreadPoolExecutor-1_0", "data": {"id": "ec9d22e8-ed36-40d2-ba6a-5d09b3f1d965", "name": "saleor.product.thumbnails.create_product_thumbnails", "return_value": "None", "runtime": 0.17312861210666597}, "hostname": "ubuntu-s-1vcpu-1gb-ams3-01"}
```

**System information**

Saleor version:
- [ ] dev (current master)
- [ x] 3.0
- [ ] 2.11
- [ ] 2.10

Operating system:
- [ ] Windows
- [ x] Linux
- [ ] MacOS
- [ ] Other

**Comment by krzysztofwolski** [comment]
Bug confirmed. ATM I dont have eta for the fix. 
Possible solution could be adding https://pypi.org/project/django-ok-images/ to the stack

**Comment by ahmedaljazzar** [comment]
Should be fixed in #7906. `django-versatileimagefield==2.2` which was added [here](https://github.com/mirumee/saleor/pull/7906/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552R40), added support for webp sizers and filters.

**Comment by patrys** [comment]
Saleor 3.6 replaces the versatile image field with custom thumbnailing code that includes support for WebP. Let's see if we can close this in a few days, once 3.6 ships.
